api/types.go

```diff
@@ -609,6 +609,59 @@ func (pins PinSerial) ToPin() Pin {
 	}
 }
 
+// AddedOutput carries information for displaying the standard ipfs output
+// indicating a node of a file has been added.
+type AddedOutput struct {
+	Error
+	Name  string
+	Hash  string `json:",omitempty"`
+	Bytes int64  `json:",omitempty"`
+	Size  string `json:",omitempty"`
+}
+
+// NodeSerial encodes info necessary to add an IPFS Block. This includes the
+// cid, data, and size
+type NodeSerial struct {
+	Cid  string
+	Data []byte
+	Size uint64
+}
+
+// ShardNodeSerial encodes the info necessary to add an IPFS Block to a sharded
+// file.  This includes the serialized node and a sharding ID.
+type ShardNodeSerial struct {
+	Cid  string
+	Data []byte
+	Size uint64
+	ID   string
+}
+
+// BlockWithFormat specifies a block of data and how it should be
+// formatted.  An empty format string will be interpreted as "v0".
+type BlockWithFormat struct {
+	Data   []byte
+	Format string
+}
+
+// AllocateInfo transports the information necessary to call an allocator's
+// Allocate function.
+type AllocateInfo struct {
+	Cid        string
+	Current    map[peer.ID]Metric
+	Candidates map[peer.ID]Metric
+}
+
+// GetCid decodes the cid string within AllocateInfo.  If the cid string is ""
+// then GetCid returns nil
+func (aI *AllocateInfo) GetCid() *cid.Cid {
+	if aI.Cid == "" {
```

Comment:
Might as well skip this check, `cid.Decode` likely returns nil for the empty string...

Reply:
Thanks, indeed it does